odl/solvers/nonsmooth/adupdates.py

```diff
@@ -0,0 +1,210 @@
+    <http://ieeexplore.ieee.org/document/7271047/>`_ is designed to solve an
+    optimization problem of the form
+
+        min_x sum_i g_i(L_i x)
+
+    where ``g_i`` are proper, convex and lower semicontinuous functions and
+    ``L_i`` are linear `Operator`s.
+
+    Parameters
+    ----------
+    g : sequence of `Functional`s
+        All functions need to provide a `convex_conj` with a
+        `proximal` factory.
+    L : sequence of `Operator`s
+        Needs to provide as many elements as `g`.
+    x : `LinearSpaceElement`
+        Initial point, updated in-place.
+    stepsize : positive float
+        The stepsize for the outer (proximal point) iteration. The theory
+        guarantees convergence for any positive real number, but the
+        performance might depend on the choice of a good stepsize.
+    majs : sequence of majorizers
+        Parameters determining the stepsizes for the inner iterations. Must be
+        matched with the norms of `L`, and convergence is guaranteed if
+        `majs` are large enough. See the Notes section for details.
+    niter : int
+        Number of (outer) iterations.
+    random : bool, optional
+        If `True`, the order of the dual upgdates is chosen randomly,
+        otherwise the order provided by the lists `g`, `L` and `majs`
+        is used.
+    callback : callable, optional
+        Function called with the current iterate after each iteration.
+    callback_loop : {'inner', 'outer'}, optional
+       If 'inner', the `callback` function is called after each inner
+       iteration, i.e., after each dual update. If 'outer', the `callback`
+       function is called after each outer iteration, i.e., after each primal
+       update.
+
+    Notes
+    -----
+    The algorithm as implemented here is described in the article [MF2015],
+    where it is applied to a tomography problem. It solves the problem
+
+    .. math::
+        \min_x \sum_{i=1}^m g_i(L_i x),
+
+    where :math:`g_i` are proper, convex and lower semicontinuous functions
+    and :math:`L_i` are linear, continuous operators for
+    :math:`i = 1, \ldots, m`. In an outer iteration, the solution is found
+    iteratively by an iteration
+
+    .. math::
+        x_{n+1} = \mathrm{arg\,min}_x \sum_{i=1}^m g_i(L_i x)
+            + \frac{\mu}{2} \|x - x_n\|^2
+
+    with some :math:`\mu > 0` according to the proximal point algorithm.
+    In the inner iteration, dual variables are introduces for each of the
+    components of the sum. The Lagrangian of the problem is given by
+
+    .. math::
+        S_n(x; v_1, \ldots, v_m) = \sum_{i=1}^m (\langle v_i, L_i x \rangle
+            - g_i^*(v_i)) + \frac{\mu}{2} \|x - x_n\|^2.
+
+    Given the dual variables, the new primal variable :math:`\tilde x_{n+1}`
+    can be calculated by directly minimizing :math:`S_n` with respect to
+    :math:`x`. This corresponds to the formula
+
+    .. math::
+         \tilde x_{n+1} = x_n - \frac{1}{\mu} \sum_{i=1}^m L_i^* v_i.
```

Comment:
What does tilde mean here?

Reply:
Remark here that applies to a lot of places in our docstrings:

If we use [raw string literals](https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals) whenever there's math in it, we can ditch all those double backslash ugliness.

Here, some math won't work (e.g. `\frac`), so the simplest fix is to just prepend the whole docstring with an `r`.

However, we should explore what happens with unicode stuff, in cases where names require non-ASCII characters.

Reply:
Unicode seems to work fine at least in my environment with Python 3.6.